--- asterisk-1.6.2.11/main/dns.c	2008-11-29 18:57:39.000000000 +0100
+++ asterisk-1.6.2.11.uclibc_patch/main/dns.c	2010-06-05 09:12:24.000000000 +0200
@@ -288,7 +288,7 @@ int ast_search_dns(void *context,
 	res_nclose(&dnsstate);
 #endif
 #else
-#ifndef __APPLE__
+#if !defined(__APPLE__) && !(defined(__UCLIBC__) && !defined(__UCLIBC_HAS_BSD_RES_CLOSE__))
 	res_close();
 #endif
 	ast_mutex_unlock(&res_lock);
