--- asterisk-1.6.2.11.orig/contrib/scripts/safe_asterisk	2010-01-12 04:25:20.000000000 +0100
+++ asterisk-1.6.2.11/contrib/scripts/safe_asterisk	2010-07-05 01:19:08.284866195 +0200
@@ -1,15 +1,14 @@
-#!/bin/sh
+#!/bin/bash
 # vim:textwidth=80:tabstop=4:shiftwidth=4:smartindent:autoindent
 
 CLIARGS="$*"				# Grab any args passed to safe_asterisk
-TTY=9						# TTY (if you want one) for Asterisk to run on
-CONSOLE=yes					# Whether or not you want a console
+#TTY=9						# TTY (if you want one) for Asterisk to run on
+CONSOLE=no					# Whether or not you want a console
 #NOTIFY=ben@alkaloid.net	# Who to notify about crashes
 #EXEC=/path/to/somescript	# Run this command if Asterisk crashes
 #LOGFILE=/path/to/logfile	# Where to place the normal logfile (disabled if blank)
 #SYSLOG=local0				# Which syslog facility to use (disabled if blank)
 MACHINE=`hostname`			# To specify which machine has crashed when getting the mail
-DUMPDROP=/tmp
 SLEEPSECS=4
 ASTSBINDIR=__ASTERISK_SBIN_DIR__
 ASTVARRUNDIR=__ASTERISK_VARRUN_DIR__
@@ -19,9 +18,6 @@
 # asterisk exits
 KILLALLMPG123=1
 
-# run asterisk with this priority
-PRIORITY=0
-
 # set system filemax on supported OSes if this variable is set
 # SYSMAXFILES=262144
 
@@ -97,11 +93,6 @@
 fi
 
 #
-# Let Asterisk dump core
-#
-ulimit -c unlimited
-
-#
 # Don't fork when running "safely"
 #
 ASTARGS=""
@@ -119,10 +110,6 @@
 		ASTARGS="${ASTARGS} -c"
 	fi
 fi
-if test ! -w "${DUMPDROP}" ; then	
-	message "Cannot write to ${DUMPDROP}"
-	exit 1
-fi
 
 #
 # Don't die if stdout/stderr can't be written to
@@ -148,10 +135,10 @@
 		if test "x$TTY" != "x" ; then
 			cd /tmp
 			stty sane < /dev/${TTY}
-			nice -n $PRIORITY ${ASTSBINDIR}/asterisk -f ${CLIARGS} ${ASTARGS} > /dev/${TTY} 2>&1 < /dev/${TTY}
+			${ASTSBINDIR}/asterisk -f ${CLIARGS} ${ASTARGS} > /dev/${TTY} 2>&1 < /dev/${TTY}
 		else
 			cd /tmp
-			nice -n $PRIORITY ${ASTSBINDIR}/asterisk -f ${CLIARGS} ${ASTARGS}
+			${ASTSBINDIR}/asterisk -f ${CLIARGS} ${ASTARGS}
 		fi
 		EXITSTATUS=$?
 		message "Asterisk ended with exit status $EXITSTATUS"
@@ -170,24 +157,8 @@
 			if test "x$EXEC" != "x" ; then
 				$EXEC
 			fi
-
-			PID=`cat ${ASTPIDFILE}`
-			DATE=`date "+%Y-%m-%dT%H:%M:%S%z"`
-			if test -f /tmp/core.${PID} ; then
-				mv /tmp/core.${PID} ${DUMPDROP}/core.`hostname`-$DATE &
-			elif test -f /tmp/core ; then
-				mv /tmp/core ${DUMPDROP}/core.`hostname`-$DATE &
-			fi
 		else
 			message "Asterisk died with code $EXITSTATUS."
-
-			PID=`cat ${ASTPIDFILE}`
-			DATE=`date "+%Y-%m-%dT%H:%M:%S%z"`
-			if test -f /tmp/core.${PID} ; then
-				mv /tmp/core.${PID} ${DUMPDROP}/core.`hostname`-$DATE &
-			elif test -f /tmp/core ; then
-				mv /tmp/core ${DUMPDROP}/core.`hostname`-$DATE &
-			fi
 		fi
 		message "Automatically restarting Asterisk."
 		sleep $SLEEPSECS
