--- chan_capi-1.1.5/Makefile	2010-04-06 19:33:25.000000000 +0200
+++ chan_capi-1.1.5.uclibc_patch/Makefile	2010-06-05 15:57:54.000000000 +0200
@@ -18,15 +18,14 @@
 
 OSNAME=${shell uname}
 
-USE_OWN_LIBCAPI=yes
+USE_OWN_LIBCAPI=no
 
 .EXPORT_ALL_VARIABLES:
 
 V=0
 
-INSTALL_PREFIX=
 
-ASTERISK_HEADER_DIR=$(INSTALL_PREFIX)/usr/include
+ASTERISK_HEADER_DIR=$(INSTALL_PREFIX)/include
 
 ifeq (${OSNAME},FreeBSD)
 ASTERISK_HEADER_DIR=$(INSTALL_PREFIX)/usr/local/include
@@ -36,7 +35,7 @@ ifeq (${OSNAME},NetBSD)
 ASTERISK_HEADER_DIR=$(INSTALL_PREFIX)/usr/pkg/include
 endif
 
-MODULES_DIR=$(INSTALL_PREFIX)/usr/lib/asterisk/modules
+MODULES_DIR=$(INSTALL_PREFIX)/lib/asterisk/modules
 
 ifeq (${OSNAME},FreeBSD)
 MODULES_DIR=$(INSTALL_PREFIX)/usr/local/lib/asterisk/modules
@@ -102,7 +101,7 @@ CFLAGS+=$(shell if $(CC) -march=$(PROC) 
 CFLAGS+=$(shell if uname -m | grep -q "ppc\|arm\|s390"; then echo "-fsigned-char"; fi)
 
 LIBS=-ldl -lpthread -lm
-CC=gcc
+CC=mipsel-linux-gcc
 INSTALL=install
 
 SHAREDOS=chan_capi.so
